lib/src/api/mod.rs

```diff
@@ -109,6 +115,8 @@ where
 					client = Client::connect(endpoint, self.capacity).await?;
 				}
 			}
+			// Both ends of the channel are still alive at this point
+			client.waiter.0.send(Some(WaitFor::Connection)).ok();
```

Comment:
If any error occurs WaitFo::Connection will never be sent.
Is there is risk for the caller to wait indefinitely?
Shall we also have a WaitFor::Error event?

Reply:
Here is a [snippet from the docs](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Sender.html#method.send):

> This method fails if the channel is closed, which is the case when every receiver has been dropped.

We store both the sender and receiver in that `waiter` field so we know at this point that they haven't been dropped. That's why it's safe to throw away the error here. That's what the comment above is trying to explain.